apps/etl-func/src/adapters/functions/messages-ingestion.ts

```diff
@@ -11,16 +11,22 @@ const logger = pino({
   level: process.env.NODE_ENV === "production" ? "error" : "debug",
 });
 
+const processMessageMetadata = (
+  input: unknown,
+): MessageMetadata | undefined => {
+  const { data, success } = messageMetadataSchema.safeParse(input);
+  return success && !data?.isPending ? data : undefined;
+};
+
 const messagesIngestionHandler =
   (ingestUseCase: IngestMessageUseCase): CosmosDBHandler =>
   async (documents: unknown[]) => {
     //Avoiding all documents different from MessageMetadata schema and with
     //isPending equals to true
-    const messagesMetadata: MessageMetadata[] = documents.filter(
-      (item): item is MessageMetadata => {
-        const parsedItem = messageMetadataSchema.safeParse(item);
-        return parsedItem.success && !parsedItem.data.isPending;
-      },
+
+    const parsedMessagesMetadata = documents.map(processMessageMetadata);
+    const messagesMetadata: MessageMetadata[] = parsedMessagesMetadata.filter(
+      (item): item is MessageMetadata => item !== undefined,
```

Comment:
Can't we unify the code like this by removing the function and linking the map and the filter together?
```suggestion
const messagesMetadata: MessageMetadata[] = documents
  .map((input) => {
    const { data, success } = messageMetadataSchema.safeParse(input);
    return success && !data?.isPending ? data : undefined;
  })
  .filter((item): item is MessageMetadata => item !== undefined);
```

Reply:
I think in this way we are going to loose readability. Do you see any problem if we keep the current version ?

Reply:
imho the cleanest solution is using `zod` to perform this validation

something like

```ts
const messagesMetadata = z.array(messageMetadataSchema).parse(documents)
```

(maybe, if needed, combined with something like zod's `.refine`)

Reply:
@lucacavallaro this way, we are unable to distinguish valid elements within the array. Our goal is to ignore invalid elements, but using this type of validation will result in the entire array being treated as invalid.

Reply:
f4fcffeb59c1b83bd3107b693ad586d5be2b572c